### PR TITLE
Bump docker base images

### PR DESCRIPTION
**Pull request by yorinasub17**
Amazon linux: 2017.12 => 2
Ubuntu: 18.04 => 20.04

### CODE REVIEW

**Review by infraredgirl — DISMISSED** [review]

File: test-docker-images/gruntwork-amazon-linux-test/Dockerfile [hunk 1]
```diff
@@ -13,21 +13,16 @@ RUN yum update -y && \
...
+        pip3 \
         wget && \
         yum clean all && rm -rf /var/cache/yum
 
-# Installing pip with yum doesn't actually put it in the PATH, so we use easy_install instead. Pip will now be placed
-# in /usr/local/bin, but amazonlinux's sudo uses a sanitzed PATH that does not include /usr/local/bin, so we symlink pip.
-# The last line upgrades pip to the latest version.
-RUN curl https://bootstrap.pypa.io/ez_setup.py | sudo /usr/bin/python && \
-    easy_install pip && \
-    pip install --upgrade pip
```
Comment (yorinasub17) [inline test-docker-images/gruntwork-amazon-linux-test/Dockerfile:24]: This is no longer necessary if you install py3

File: test-docker-images/gruntwork-amazon-linux-test/Dockerfile [hunk 2]
```diff
@@ -13,21 +13,16 @@ RUN yum update -y && \
...
+# Upgrade pip and install awscli
+RUN pip3 install --upgrade pip && pip3 install awscli
 
 # Ideally, we'd install the latest version of Docker to avoid a conflict between the Docker client in this container
 # and the Docker API on your local host, but installing the latest version of Docker yields the error "Requires:
 # container-selinux >= 2.9", whch indicates that a newer Linux kernel version is required than what comes with Amazon Linux.
 # So we settle for the Amazon Linux supported version for now.
-RUN yum install -y docker
\ No newline at end of file
+RUN amazon-linux-extras install docker
```
Comment (yorinasub17) [inline test-docker-images/gruntwork-amazon-linux-test/Dockerfile:28]: New way to install docker in AL2

**Comment by yorinasub17** [comment]
@brikis98 FYI I just realized I don't have access to the docker hub repos to push these images. Is it possible to grant me access, or grant the `gruntworkci` user access?

**Comment by yorinasub17** [comment]
Closing as stale

**Pull request closed without merging**